dvc/repo/checkout.py

```diff
@@ -30,11 +31,14 @@ def checkout(self, target=None, with_deps=False, force=False, recursive=False):
     with self.state:
         _cleanup_unused_links(self, all_stages)
 
-        for stage in stages:
+        for index, stage in enumerate(stages):
             if stage.locked:
                 logger.warning(
                     "DVC file '{path}' is locked. Its dependencies are"
                     " not going to be checked out.".format(path=stage.relpath)
                 )
 
             stage.checkout(force=force)
+            progress.update_target(
+                "Checkout in progress", index + 1, len(stages)
+            )
```

Comment:
I am not sure how to test it. If you want to see how does it work, put `time.sleep(0.5)` after `progress.update_target`
and run following script:
```
#!/bin/bash

rm -rf repo
rm -rf storage

mkdir repo
mkdir storage

cd repo

git init
dvc init

for i in {1..25}
do
   echo $i >> $i 
   dvc add $i
done

dvc remote add -d storage ../storage

dvc push

echo "###########Removing!"
for i in {1..25}
do
   rm $i
done	

dvc pull 

```

Reply:
Mock update_target and check that it has been called with a certain combination of arguments?

Reply:
Hmm, that might be good idea, but maybe instead of update, mock logger and check whether propper methods have been called with proper arguments and order?

Reply:
It depends on what are you actually testing. If you the idea is to test `checkout` I would mock `progress` (and may be logger if it's called directly from the `checkout` implementation).

Reply:
I think that in this case call order and aruments are important (mostly because checkout is tested) and because we want to make sure that there won't be dangling partial progress bars in log.